lib/runner/extensions/event.command.js

```diff
@@ -398,12 +393,27 @@ module.exports = {
                         }
                     }.bind(this));
 
-                this.host.on(EXECUTION_VAULT_BASE + executionId, function (id, cmd, ...args) {
+                this.host.on(EXECUTION_VAULT_BASE + executionId, async function (id, cmd, ...args) {
+                    try {
+                        if (typeof isVaultAccessInScriptsAllowed === 'function') {
+                            isVaultAccessAllowed = await isVaultAccessInScriptsAllowed(item.id);
```

Comment:
```suggestion
                            isVaultAccessAllowed = Boolean(await _.get(vaultSecrets, '_.allowScriptAccess', _.noop)(item.id));
```

Reply:
You can clean up the code above (remove if condition and unused vars) accrodingly.

Reply:
Also, we don't need to run the consent check everytime there is get/set/unset on vault secrets. It should just be done for the first time in that execution, unless we are allowing the conset to be changed mid-execution (IMO that would be just confusing for users)?